evals/sampling_eval.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import seaborn as sns
+import torch.nn as nn
+import matplotlib.pyplot as plt
+from omegaconf import OmegaConf
+from huggingface_hub.file_download import hf_hub_download
+
+from sample import Generator
+from train import preprocess_dataset
+from models.reverse_diffusion import Unet
+from models.forward_diffusion import ForwardDiffusion
+from models.velocity_time_encoder import VelocityTimeEncoder
+
+
+def eval_generation(
+    gen: Generator,
+    cfg: OmegaConf,
+    conditioning_model: nn.Module = None,
+    classifier_free_guidance_scale: float = 3.0,
+    batch_size: int = 2,
+):
+    _, loader, _ = preprocess_dataset("JasiekKaczmarczyk/giant-midi-sustain-quantized", batch_size, 1, overfit_single_batch=False)
+
+    batch = next(iter(loader))
```

Comment:
```suggestion
    _, loader, _ = preprocess_dataset(
        dataset_name="JasiekKaczmarczyk/giant-midi-sustain-quantized",
        batch_size=batch_size,
        ???=1,
        overfit_single_batch=False,
    )

    batch = next(iter(loader))
```

Please use named parameters to prevent reader confusion :)

Also, I think we shouldn't have to create a dataloader and a single-element batch in order to run validations. Ideally we could use a dataset class and get a single record for processing 🤔